lib/stellar/transaction_builder.rb

```diff
@@ -0,0 +1,108 @@
+module Stellar
+  class TransactionBuilder
+
+    attr_reader :source_account, :sequence_number, :base_fee, :time_bounds, :time_bounds, :memo, :operations
+
+    def initialize(
+      source_account:, 
+      sequence_number:, 
+      base_fee: 100, 
+      time_bounds: nil, 
+      memo: nil
+    )
+      raise ArgumentError, "bad :source_account" unless source_account.is_a?(Stellar::KeyPair)
+      raise ArgumentError, "bad :sequence_number" unless sequence_number.is_a?(Integer) && sequence_number > 0
+      raise ArgumentError, "bad :time_bounds" unless time_bounds.is_a?(Stellar::TimeBounds) || time_bounds.nil?
+      raise ArgumentError, "bad :base_fee" unless base_fee.is_a?(Integer) && base_fee >= 100
```

Comment:
Can we start these errors with an upper case letter? There are two places in this gem that we start argument errors with a lower case, but the vast majority of cases we start them with an upper. Great to be consistent.
```suggestion
      raise ArgumentError, "Bad :source_account" unless source_account.is_a?(Stellar::KeyPair)
      raise ArgumentError, "Bad :sequence_number" unless sequence_number.is_a?(Integer) && sequence_number > 0
      raise ArgumentError, "Bad :time_bounds" unless time_bounds.is_a?(Stellar::TimeBounds) || time_bounds.nil?
      raise ArgumentError, "Bad :base_fee" unless base_fee.is_a?(Integer) && base_fee >= 100
```
Can we do the same with all the other errors we're raising in this PR?